system/kube-system/values.yaml

```diff
@@ -1,2 +1,120 @@
-ingress:
-  external_service_ip: DEFINED-AS-VALUE
+nginx-ingress:
+  revisionHistoryLimit: 5
+
+  controller:
+    image:
+      repository: sapcc/nginx-ingress-controller
+      tag: "0.12.0"
+
+    config:
+      http-snippet: |
+        map $ssl_client_s_dn $ssl_client_s_dn_cn {
+            default "anonymous";
+            ~CN=(?<CN>[^/,\"]+) $CN;
+        }
+      location-snippet: |
+        proxy_set_header X-REMOTE-USER $ssl_client_s_dn_cn;
+      proxy-read-timeout: '300'
+      proxy-body-size: '0'
+      server-name-hash-bucket-size: '128'
+      map-hash-bucket-size: '128'
+      worker-processes: '8'
+      disable-ipv6: 'true'
+      ssl-redirect: 'false'
+
+    kind: Deployment
+    replicaCount: 3
+
+    minReadySeconds: 60
+    updateStrategy:
+      type: RollingUpdate
+      rollingUpdate:
+        maxSurge: 0
+        maxUnavailable: 1
+
+    extraArgs:
+      enable-ssl-passthrough:
+      annotations-prefix: ingress.kubernetes.io
+
+    stats:
+      enabled: true
+
+    metrics:
+      enabled: true
+      service:
+        annotations:
+          prometheus.io/scrape: "true"
+          prometheus.io/probe: "10254"
+
+    service:
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "10254"
+
+      targetPorts:
+        http: http
+        https: https
+
+      # set in regional values
+      externalIPs:
+        - DEFINED-AS-VALUE
+
+    defaultBackendService: "kube-system/ingress-default-backend"
+
+    podLabels:
+      app: nginx-ingress-controller
+
+    affinity:
```

Comment:
Hey @SchwarzM,
I'd like to replace the `hostPort` hack we're currently using to get 1 ingress-controller per master node with affinities. I remember you tried that already elsewhere. Can you take a look at this `affinity`? Thanks a lot in advance.

Reply:
lgtm